app/src/main/java/com/example/farmeraid/data/MarketRepository.kt

```diff
@@ -2,69 +2,79 @@ package com.example.farmeraid.data
 
 import android.util.Log
 import com.example.farmeraid.data.model.MarketModel
+import com.example.farmeraid.data.model.ResponseModel
+import com.google.firebase.firestore.FirebaseFirestore
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.tasks.await
 
 class MarketRepository(
-    private val quotasRepository: QuotasRepository
+    private val quotasRepository: QuotasRepository,
+    private val farmRepository: FarmRepository
 ) {
-    private val markets: MutableList<MarketModel.Market> = mutableListOf(
-        MarketModel.Market(id = "NanrSSGcmdsG2nYwbZRu", name = "St. Jacob's"),
-        MarketModel.Market(id = "NanrSSGcmdsG2nYwbZRu", name = "St. Lawrence"),
-        MarketModel.Market(id = "NanrSSGcmdsG2nYwbZRu",name = "Kensington Market"),
-        MarketModel.Market(id = "NanrSSGcmdsG2nYwbZRu",name = "St. Catherines Market"),
-        MarketModel.Market(id = "NanrSSGcmdsG2nYwbZRu", name = "Test Market"),
-    )
+    private var db: FirebaseFirestore = FirebaseFirestore.getInstance()
+    private suspend fun readMarketData(): ResponseModel.FAResponseWithData<List<MarketModel.Market>> {
+        val marketIds = farmRepository.getMarketIds()
+        val marketModel = marketIds.data?.map { element ->
+            db.collection("market").document(element).get().await()
+        }?.toList()
 
-    fun getMarkets(): Flow<List<MarketModel.Market>> {
+        val marketModelList = mutableListOf<MarketModel.Market>()
+
+        if (marketModel != null) {
+            for (market in marketModel) {
+                Log.e("ID", market.id)
+                marketModelList.add(MarketModel.Market(
+                    id = market.id,
+                    name = market.get("name") as String,
+                    quotaID = market.get("quota_id") as String
+                ))
+            }
+        }
+
+        return ResponseModel.FAResponseWithData.Success(marketModelList as List<MarketModel.Market>)
```

Comment:
do you need the `as List<MarketModel.Market>` here? I don't think you need it if you're going from mutable to non-mutable but just confirm that when you make the change.